### PR TITLE
Add tests and clean up package

### PR DESCRIPTION
**Pull request by thomvet**
- This PR adds the standard Aqua.jl tests, introduces a continuous integration workflow (CI.yml) and finally restructures the files in the test folder into a set of reference tests.
- Introducing the Aqua tests highlighted a few cases of type piracy; I removed these by introducing small helper functions, rather than extending functionality from LinearAlgebra and Base onto types that this package doesn't own.
- There was also a bug in the compat entries for Makie (0.2 instead of 0.20 - not the same thing in semver!), but the bug was only on master, not released.
- The CI.yml workflow is untested. I hope it works, but if not, I am happy to tweak it.

I hope the changes will be seen as useful.

### CODE REVIEW

**Comment by theogf** [comment]
That's really nice @stelmo could this get merged?

**Comment by stelmo** [comment]
Hi all, sorry for the super long delay! I was on holiday and swamped with other stuff! Merging straight away :)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@v2
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v2
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+        with:
+          prefix: DISPLAY=:0 xvfb-run -s '-screen 0 1024x768x24' #"headless", i.e., without display, as CI server doesn't have a monitor
```
Comment (theogf) [inline .github/workflows/CI.yml:36]: It looks like this command broke the CI

Comment (theogf) [reply .github/workflows/CI.yml:36]: Also `xvfb-run` does not seem to exist on MacOS

(or Windows)

Comment (theogf) [reply .github/workflows/CI.yml:36]: Trying fixes in https://github.com/stelmo/TernaryDiagrams.jl/pull/21

**Pull request merged**